src/seedu/addressbook/ui/Formatter.java

```diff
@@ -0,0 +1,103 @@
+package seedu.addressbook.ui;
+
+import static seedu.addressbook.common.Messages.MESSAGE_GOODBYE;
+import static seedu.addressbook.common.Messages.MESSAGE_INIT_FAILED;
+import static seedu.addressbook.common.Messages.MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE;
+import static seedu.addressbook.common.Messages.MESSAGE_USING_STORAGE_FILE;
+import static seedu.addressbook.common.Messages.MESSAGE_WELCOME;
+
+import java.io.PrintStream;
+import java.util.List;
+
+/**
+ * Formats text for display (e.g. adding decorations)
+ */
+public class Formatter {
+
+    /** A decorative prefix added to the beginning of lines printed by AddressBook */
+    private static final String LINE_PREFIX = "|| ";
+
+    /** A platform independent line separator. */
+    private static final String LS = System.lineSeparator();
+
+    private static final String DIVIDER = "===================================================";
+
+    /** Format of indexed list item */
+    private static final String MESSAGE_INDEXED_LIST_ITEM = "\t%1$d. %2$s";
+
+    /** Offset required to convert between 1-indexing and 0-indexing.  */
+    public static final int DISPLAYED_INDEX_OFFSET = 1;
+
+    private final PrintStream out;
+
+    public Formatter(PrintStream out) {
+        this.out = out;
+    }
+
+    public void getCommandPrompt() {
+        out.print(LINE_PREFIX + "Enter command: ");
+    }
+
+    public void showToUserCommand(String fullInputLine) {
+        showToUser("[Command entered:" + fullInputLine + "]");
+    }
+
+    public void showToUserWelcomeMessage(String version, String storageFilePath) {
+        String storageFileInfo = String.format(MESSAGE_USING_STORAGE_FILE, storageFilePath);
+        showToUser(
+                DIVIDER,
+                DIVIDER,
+                MESSAGE_WELCOME,
+                version,
+                MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE,
+                storageFileInfo,
+                DIVIDER);
+    }
+
+    public void showToUserGoodbyeMessage() {
+        showToUser(MESSAGE_GOODBYE, DIVIDER, DIVIDER);
+    }
+
+    public void showToUserInitFailedMessage() {
+        showToUser(MESSAGE_INIT_FAILED, DIVIDER, DIVIDER);
+    }
+
+    public void showToUserResult(String feedbackToUser) {
+        showToUser(feedbackToUser, DIVIDER);
+    }
+
+    /**
+     * Shows message(s) to the user
+     */
+    public void showToUser(String... message) {
+        for (String m : message) {
+            out.println(LINE_PREFIX + m.replace("\n", LS + LINE_PREFIX));
```

Comment:
Your formatter is doing both format and print. This defeats the purpose of segregating the 2 classes.